File: MATEMÁTICA/1199/main.dart
import 'dart:io';

void main() {
  while (true) {
    String number = stdin.readLineSync()!;
    if (number.contains('x')) {
      print(int.parse(number));
    } else if (int.parse(number) > -1) {
      print('0x${int.parse(number).toRadixString(16).toUpperCase()}');
    } else {
      break;
    }
  }
}

File: MATEMÁTICA/1199/main.js
let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');

while (true) {
    let number = lines.shift();
    if (number.includes('x')) {
        console.log(parseInt(number));
    } else if (parseInt(number) > -1) {
        console.log(`0x${parseInt(number).toString(16).toUpperCase()}`);
    } else {
        break;
    }
}
